### ISSUE TITLE
Use Xdebug for application development in PhpStorm

### ISSUE THREAD

**Issue by pli888** [gigascience/gigadb-website#1323]
## User story

>As a developer  
>I want to use a debugging tool in my development work
>So that I can improve my PHP development experience

## Acceptance criteria

>Given we use Docker and PhpStorm to develop PHP-based applications and tools
>When I use PhpStorm IDE to write PHP code  
>Then I can use Xdebug installed within Docker containers to step through my PHP code within PhpStorm

### Learning material

* https://docs.moodle.org/dev/Setting_up_PhpStorm
* https://nicoverbruggen.be/blog/xdebug-with-valet-and-phpstorm
* https://www.youtube.com/watch?v=GokeXqI93x8
* https://www.jetbrains.com/help/phpstorm/2023.1/configuring-xdebug.html?utm_source=product&utm_medium=link&utm_campaign=PS&utm_content=2023.1#configuring-xdebug-for-jit
* https://medium.com/@nicolas.valverde/phpstorm-docker-ccc4ce9a0b8e
* https://medium.com/the-sensiolabs-tech-blog/how-to-use-xdebug-in-docker-phpstorm-76d998ef2534
* https://hackernoon.com/how-to-debug-php-container-with-xdebug-and-phpstorm-1b2k3yjo
* https://matthewsetter.com/setup-step-debugging-php-xdebug3-docker/
* https://enrise.com/2018/02/debugging-php-with-xdebug/
* https://wpml.org/tutorials/2023/06/local-wordpress-development-with-docker-phpstorm-xdebug/

## Product Backlog Item Ready Checklist

* [ ] Business value is clearly articulated
* [ ] Item is understood enough by the IT team so it can make an informed decision as to whether it can complete this item
* [ ] Dependencies are identified and no external dependencies would block this item from being completed
* [ ] At the time of the scheduled sprint, the IT team has the appropriate composition to complete this item
* [ ] This item is estimated and small enough to comfortably be completed in one sprint
* [ ] Acceptance criteria are clear and testable
* [ ] Performance criteria, if any, are defined and testable
* [ ] The Scrum team understands how to demonstrate this item at the sprint review

## Product Backlog Item Done Checklist

* [ ] Item(s) in increment pass all Acceptance Criteria
* [ ] Code is refactored to best practices and coding standards
* [ ] Documentation is updated as needed
* [ ] Data security has not been compromised (with particular reference to the personal information we hold in GigaDB)
* [ ] No deviation from the team technology stack and software architecture has been introduced
* [ ] The product is in a releasable state (i.e. the increment has not broken anything)

**Comment by rija** [comment]
Hi @alli83 

did you say that you were working on enabling xdbug in one of your current legacy tech debts repayment PR ?

**Comment by alli83** [comment]
@rija yes, but it's Xdebug for coverage. I also use Xdebug locally with phpStorm. If you want I can create a separate PR
